tensorflow_addons/metrics/matthews_correlation_coefficient.py

```diff
@@ -0,0 +1,152 @@
+        super(MatthewsCorrelationCoefficient, self).__init__(
+            name=name, dtype=dtype)
+        self.num_classes = num_classes
+        self.true_positives = self.add_weight(
+            'true_positives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.false_positives = self.add_weight(
+            'false_positives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.false_negatives = self.add_weight(
+            'false_negatives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+        self.true_negatives = self.add_weight(
+            'true_negatives',
+            shape=[self.num_classes],
+            initializer='zeros',
+            dtype=self.dtype)
+
+    # TODO: sample_weights
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        self.sample_weight = sample_weight
+        y_true = tf.cast(y_true, tf.float32)
+        y_pred = tf.cast(y_pred, tf.float32)
+
+        true_positive = tf.math.count_nonzero(y_true * y_pred, 0)        
+        # true_negative   
+        y_true_negative = tf.math.not_equal(y_true,1.0)
+        y_pred_negative = tf.math.not_equal(y_pred,1.0)
+        true_negative = tf.math.count_nonzero(tf.math.logical_and(y_true_negative,y_pred_negative), axis=0)
+        # predicted sum
+        pred_sum = tf.math.count_nonzero(y_pred, 0)
+        # Ground truth label sum
+        true_sum = tf.math.count_nonzero(y_true, 0)
+        false_positive = pred_sum - true_positive
+        false_negative = true_sum - true_positive
+        
+        # true positive state_update
+        self.true_positives.assign_add(tf.cast(true_positive, self.dtype))
+        # false positive state_update
+        self.false_positives.assign_add(tf.cast(false_positive, self.dtype))
+        # false negative state_update
+        self.false_negatives.assign_add(tf.cast(false_negative, self.dtype))
+        # true negative state_update
+        self.true_negatives.assign_add(tf.cast(true_negative, self.dtype))
+
+    def result(self):
+        # numerator
+        numerator1 = self.true_positives * self.true_negatives
+        numerator2 = self.false_positives * self.false_negatives
+        numerator = numerator1 - numerator2
+        # denominator
+        denominator1 = self.true_positives + self.false_positives
+        denominator2 = self.true_positives + self.false_negatives
+        denominator3 = self.true_negatives + self.false_positives
+        denominator4 = self.true_negatives + self.false_negatives
+        denominator = tf.math.sqrt(
+            denominator1 * denominator2 * denominator3 * denominator4)
+        mcc = tf.math.divide_no_nan(numerator, denominator)
+        return mcc
+
+    def get_config(self):
+        """Returns the serializable config of the metric."""
+
```

Comment:
ditto

Reply:
Following other metrics, they keep an empty line here. What do you think?